trackintel/analysis/location_identification.py

```diff
@@ -0,0 +1,229 @@
+    location as the home location is derived from geo-tagged tweets. We directly sum the time
+    spent at a location as our data model includes that.
+
+    References
+    ----------
+    Chen, Qingqing, and Ate Poorthuis. 2021.
+    ‘Identifying Home Locations in Human Mobility Data: An Open-Source R Package for Comparison and Reproducibility’.
+    International Journal of Geographical Information Science 0 (0): 1–24.
+    https://doi.org/10.1080/13658816.2021.1887489.
+
+    Examples
+    --------
+    >>> ti.analysis.location_identifier(sps, pre_filter=True, recipe="FREQ")
+    """
+    assert sps.as_staypoints
+    sps.copy()
+    if "location_id" not in sps.columns:
+        raise KeyError(
+            (
+                "To derive activity labels the GeoDataFrame (as trackintel staypoints) must have a column "
+                f"named 'location_id' but it has [{', '.join(sps.columns)}]"
+            )
+        )
+    if pre_filter:
+        f = pre_filter_locations(sps, **pre_filter_kwargs)
+    else:
+        f = pd.Series([True for _ in sps.index])
+
+    if recipe == "FREQ":
+        recipe_val = freq_recipe(sps[f], "home", "work")
+    else:
+        raise ValueError(f"Recipe {recipe} does not exist.")
+
+    sps.loc[f, "activity_label"] = recipe_val["activity_label"]
+    return sps
+
+
+def pre_filter_locations(
+    sps,
+    agg_level="user",
+    thresh_sp=10,
+    thresh_loc=10,
+    thresh_sp_at_loc=10,
+    thresh_loc_time="1h",
+    thresh_loc_period="5h",
+):
+    """Filter locations and user out that have not enough data to do a proper analysis.
+
+    To disable a specific filter parameter set it to zero.
+
+    Parameters
+    ----------
+    sps : GeoDataFrame (as trackintel staypoints)
+        Staypoints with the column "location_id".
+
+    agg_level: {"user", "dataset"}, default "user"
+        The level of aggregation when filtering locations:
+        - 'user' : locations are filtered per-user.
+        - 'dataset' : locations are filtered over the whole dataset.
+
+    thresh_sp : int, default 10
+        Minimum staypoints a user must have to be included.
+
+    thresh_loc : int, default 10
+        Minimum locations a user must have to be included.
+
+    thresh_sp_at_loc : int, default 10
+        Minimum number of staypoints at a location must have to be included.
+
+    thresh_loc_time : str or pd.Timedelta, default "1h"
+        Minimum sum of durations that was spent at location to be included.
+        If str must be parsable by pd.to_timedelta.
+
+    thresh_loc_period : str or pd.Timedelta, default "5h"
+        Minimum timespan of first to last visit at a location to be included.
+        If str must be parsable by pd.to_timedelta.
+
+    Returns
+    -------
+    pd.Series
+        Boolean series containing the filter as a mask.
+
+    Examples
+    --------
+    >> sps = sps[ti.analysis.pre_filter_locations(sps)]
+    """
+    assert sps.as_staypoints
+    sps = sps.copy()
```

Comment:
we do not need this now, as spt is already copied in the main function?

Reply:
Yes, but it should be possible to access the function independently. Maybe that's a bit too much.